mri/operators/fourier/utils.py

```diff
@@ -286,3 +284,29 @@ def gridded_inverse_fourier_transform_stack(kspace_data_sorted,
     # Transpose every image in each slice
     return np.swapaxes(np.fft.fftshift(np.fft.ifftn(np.fft.ifftshift(
         gridded_kspace))), 0, 1)
+
+
+def check_if_fourier_op_uses_sense(fourier_op):
+    """Utils function to check if fourier operator uses SENSE recon
+
+    Parameters
+    ----------
+
+    fourier_op: object of class FFT, NonCartesianFFT or Stacked3DNFFT in
+    mri.operators
+        the fourier operator for which we want to check if SENSE is
+        supported
+
+    Returns
+    -------
+    bool
+        True if SENSE recon is being used
+    """
+    from .non_cartesian import NonCartesianFFT, gpuNUFFT
+    if isinstance(fourier_op, NonCartesianFFT):
+        if isinstance(fourier_op.implementation, gpuNUFFT):
```

Comment:
you can put them on the same line if you keep this order.

Reply:
No dude, I tried that, it fails for Cartesian case for some reason.
I am assuming you mean to use `and` 
In C the operands are not calculated , here it looks like it is

Reply:
I am very surprised because I thought the second operand was not if the first was false. For example if I try the following one-liner: `bool(False and list().not_exist())`, I don't have an error, just `False`. 

Furthermore, if I do the following:
```python
import numpy as np

from mri.operators.fourier.cartesian import FFT
from mri.operators.fourier.non_cartesian import NonCartesianFFT, gpuNUFFT

f = FFT(samples=np.array([[0,0]]), shape=(1,1))
bool(isinstance(f, NonCartesianFFT) and isinstance(f.implementation, gpuNUFFT))
```

I just get `False` not an error. I guess there was another problem here.

Reply:
Hmmm, Lemme run this again

Reply:
Yes, you were right, earlier, I had some other issue, it is working normal as expected now.